### DISCUSSION TITLE
Option in Server Pro Console to change a client's ID

### DISCUSSION THREAD

**Discussion by Koruudo** [rustdesk/rustdesk-server-pro#225] [General]
I checked around the discussions here and couldn't find anything regarding this, so I figured I'd post in just in case.

An option in the RustDesk Pro Console to change a client's ID would be great. I prefer the ID to be the computer's hostnames, which is their Dell Service Tag (i.e. 1J4K023), and then in the Device Name column I type the employee's name who uses it.

Previously with the standard installer, I would use this PowerShell block in the deployment script to rename the ID to the computer's hostname:
> # Change ID to Hostname
> try
> {
>     Write-Output "Attempting to change ID to hostname..."
>     $Name = $env:COMPUTERNAME
>     $id = Get-Content C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\config\RustDesk.toml | Select -First 1
>     $newId = "id = '$Name'"
>     $filecontent = Get-Content -Path C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\config\RustDesk.toml -Raw
>     $filecontent.Replace("$id","$newId") | Set-Content -Path C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\config\RustDesk.toml
>     Write-Output "ID change successful!"
> }
> 
> catch
> {
>     Write-Output "Failed trying to change ID to hostname"
> }
> 
> # Finally, restart the RustDesk service
> Restart-Service -Name "RustDesk"

But now with the custom client, this doesn't seem to work.

I appreciate any input. :)

**Comment by rustdesk** [comment]
I will test 

@fufesou it seems really not working, please help fix.

**Comment by rustdesk** [comment]
Sorry, it works.

![image](https://github.com/rustdesk/rustdesk/assets/71636191/bbac8118-a059-4e60-8f44-f2d525ef631b)

**Reply by Koruudo** [reply]
I'm guessing there's some software limitation to having the ID start with a number? Just disappointed cause some of our Dell laptops' service tags start with numbers 😅

**Comment by DMSnz** [comment]
Your subject line suggests using the Console to rename.
I am on 1.2.0 Server Pro and can edit a device, renaming it whatever I like.
I even just tested by using a number as the first character and it accepted this no issue.
Maybe that is enough for your use case.
![image](https://github.com/rustdesk/rustdesk-server-pro/assets/139725889/1168b4b1-28b0-4cfc-bbfd-8048403d33c5)

**Comment by olegmm** [comment]
RustDesk Server Pro 1.5.2  but I not able do this, no see any option fot ID changing

**Reply by rustdesk** [reply]
No, 1.5.2 does not have this, the id mention in change log is not this. We still have no plan for this, because this is actually a fundamental change for us. Though you see it is simple.
